vectorlm/utils/model_utils.py

```diff
@@ -27,106 +30,63 @@
 )
 
 
-def _is_bfloat_available() -> bool:
-    """
-    Return whether bfloat is supported for the
-    current CUDA device.
+def get_half_precision_model(model: nn.Module) -> nn.Module:
```

Comment:
Does this need to be a function? It seems to not do anything significant.

Reply:
The initially idea was to switch between bfloat16 and float16 depending on the GPU (it seems that not all GPUs support bfloat16), but at the moment we are using bfloat16 only because:
- LoRA didn't work well with float16.
- Most of our GPUs are compatible with bfloat16.

I will revert this back to shortly- might reopen when we need to support a GPU that isn't compatible with bfloat16.